pkg/datastoresrp/setup/setup.go

```diff
@@ -0,0 +1,167 @@
+	rds_ctrl "github.com/radius-project/radius/pkg/datastoresrp/frontend/controller/rediscaches"
+	sql_ctrl "github.com/radius-project/radius/pkg/datastoresrp/frontend/controller/sqldatabases"
+	mongo_proc "github.com/radius-project/radius/pkg/datastoresrp/processors/mongodatabases"
+	rds_proc "github.com/radius-project/radius/pkg/datastoresrp/processors/rediscaches"
+	sql_proc "github.com/radius-project/radius/pkg/datastoresrp/processors/sqldatabases"
+	pr_ctrl "github.com/radius-project/radius/pkg/portableresources/backend/controller"
+	rp_frontend "github.com/radius-project/radius/pkg/rp/frontend"
+)
+
+const (
+	// AsyncOperationRetryAfter is polling interval for async create/update or delete resource operations.
+	AsyncOperationRetryAfter = time.Duration(5) * time.Second
+)
+
+// SetupNamespace builds the namespace for datastores resource provider.
+func SetupNamespace(recipeControllerConfig *controllerconfig.RecipeControllerConfig) *builder.Namespace {
+	ns := builder.NewNamespace("Applications.Datastores")
+
+	_ = ns.AddResource("redisCaches", &builder.ResourceOption[*datamodel.RedisCache, datamodel.RedisCache]{
+		RequestConverter:  converter.RedisCacheDataModelFromVersioned,
+		ResponseConverter: converter.RedisCacheDataModelToVersioned,
+
+		Put: builder.Operation[datamodel.RedisCache]{
+			UpdateFilters: []apictrl.UpdateFilter[datamodel.RedisCache]{
+				rp_frontend.PrepareRadiusResource[*datamodel.RedisCache],
+			},
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewCreateOrUpdateResource[*datamodel.RedisCache, datamodel.RedisCache](options, &rds_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ResourceClient, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Patch: builder.Operation[datamodel.RedisCache]{
+			UpdateFilters: []apictrl.UpdateFilter[datamodel.RedisCache]{
+				rp_frontend.PrepareRadiusResource[*datamodel.RedisCache],
+			},
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewCreateOrUpdateResource[*datamodel.RedisCache, datamodel.RedisCache](options, &rds_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ResourceClient, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
+			AsyncOperationRetryAfter: AsyncOperationRetryAfter,
+		},
+		Delete: builder.Operation[datamodel.RedisCache]{
+			AsyncJobController: func(options asyncctrl.Options) (asyncctrl.Controller, error) {
+				return pr_ctrl.NewDeleteResource[*datamodel.RedisCache, datamodel.RedisCache](options, &rds_proc.Processor{}, recipeControllerConfig.Engine, recipeControllerConfig.ConfigLoader)
+			},
+			AsyncOperationTimeout:    ds_ctrl.AsyncCreateOrUpdateRedisCacheTimeout,
```

Comment:
Do we need to use "AsyncDeleteRedisCacheTimeout" for delete operation
https://github.com/project-radius/radius/blob/4bda9604f26c593cc3aa3deeb965dbc08c1c3497/pkg/datastoresrp/frontend/controller/types.go#L39

Reply:
whoops yes, thought I updated these but will fix